src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1621,6 +1547,45 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 			reflectionContext.RecordHandledPattern ();
 		}
 
+		static string GetDidNotMatchRequirementsWarningMessage (IMetadataTokenProvider source, IMetadataTokenProvider target)
```

Comment:
We actually had this approach before - but intentionally switched to the current one where the message is not calculated. The reason is localization and resources. Calculated messages are VERY hard to localize. The medium term (hopefully 6.0) plan is to move all the warning/error messages to resources - in which case the "calculation" approach will not work very well.
I know it is tedious to fix 40-something message by hand, but it's what we want long-term.

Reply:
That makes sense - I was curious why we had that many hard-written messages with an easy to predict structure. I'll change this back.

Reply:
Since you tripped on this - can you please add a comment with the reasoning?

Reply:
Would this be a good time to start moving the message format to a resource? Since we're changing each of the locations anyway?

Reply:
That would be nice - but I don't know how complex it will be to setup resources for linker (it needs to build for mono as well). If it's reasonably easy, we could start migrating to resources for sure.